src/main/java/uk/gov/pay/ledger/queue/managed/QueueMessageReceiver.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.pay.ledger.queue.managed;
+
+import com.google.inject.Inject;
+import io.dropwizard.lifecycle.Managed;
+import io.dropwizard.setup.Environment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.ledger.app.LedgerConfig;
+import uk.gov.pay.ledger.app.config.QueueMessageReceiverConfig;
+import uk.gov.pay.ledger.queue.EventMessageHandler;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class QueueMessageReceiver implements Managed {
+
+    private static final String QUEUE_MESSAGE_RECEIVER_THREAD_NAME = "queue-message-receiver-%d";
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueueMessageReceiver.class);
+    private final QueueMessageReceiverConfig config;
+
+    private ScheduledExecutorService scheduledExecutorService;
+    private EventMessageHandler eventMessageHandler;
+
+    @Inject
+    public QueueMessageReceiver(
+            Environment environment,
+            LedgerConfig configuration,
+            EventMessageHandler eventMessageHandler) {
+        this.eventMessageHandler = eventMessageHandler;
+        this.config = configuration.getQueueMessageReceiverConfig();
+
+        int queueReadScheduleNumberOfThreads = config.getNumberOfThreads();
+
+        scheduledExecutorService = environment
+                .lifecycle()
+                .scheduledExecutorService(QUEUE_MESSAGE_RECEIVER_THREAD_NAME)
+                .threads(queueReadScheduleNumberOfThreads)
+                .build();
+    }
+
+    @Override
+    public void start() {
+        long initialDelay = config.getThreadDelayInMilliseconds();
+        long delay = config.getThreadDelayInMilliseconds();
+
+        scheduledExecutorService.scheduleWithFixedDelay(
+                this::receive,
+                initialDelay,
+                delay,
+                TimeUnit.MILLISECONDS
```

Comment:
Is there any reason we have MILLISECONDS unit ? We have got SECONDS almost everywhere. Feel like we will mix of two units

Reply:
It's because the delay is supposed to be very small (less than a second).

Reply:
In that case we need to make sure we configure it to be less frequent in the environment we don't want (local dev environment)